### PR TITLE
Fixed: ?? didn't work here as bool won't be seen as null

### PR DESCRIPTION
**Pull request by Quix0r**
`??` didn't catch `false` here, needs to check explicitly because the returned type is always `array`.

PS: I just checked all occurrences. And some needed fixing.

### CODE REVIEW

File: src/Content/Text/BBCode.php [hunk 1]
```diff
@@ -111,7 +111,7 @@ private static function getOldAttachmentData($body)
 
 					$picturedata = Images::getInfoFromURLCached($matches[1]);
 
-					if ($picturedata) {
+					if (!empty($picturedata)) {
```
Comment (MrPetovan) [inline src/Content/Text/BBCode.php]: Empty array gets casted as false in a condition, what is the purpose of `empty()` here?

Comment (Quix0r) [reply src/Content/Text/BBCode.php]: I normally don't like this "magic casting" thing and try to make it clean (with no casting). Plus also in other occurrences I found the `empty($array)` calls. Should I then rewrite all to one way?

Comment (MrPetovan) [reply src/Content/Text/BBCode.php]: It's not wrong, just superfluous regarding the basic capabilities of the language. I'd say leave existing instances alone but don't add any more.

Comment (Quix0r) [reply src/Content/Text/BBCode.php]: I would like to make each handling of returned variables from `Imagess::getInfoFromURL[Cached]()` the same way, that we know this is everywhere the same and don't have to check each case.

Comment (Quix0r) [reply src/Content/Text/BBCode.php]: I have now removed `empty()` from each check to make it be the same on all such occurrences.

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**